### ISSUE TITLE
Question about parallel decoding

### ISSUE THREAD

**Issue by niiickZ** [moojink/openvla-oft#27]
Thanks for your great work. I have some questions regarding the implementation of parallel decoding.
- My understanding is that only the action chunk output is modified to bidirectional attention, while the image and instruction parts still use causal masks. However, in the [forked transformers](https://github.com/huggingface/transformers/commit/bc339d9ad707454c0c115970db43c260067c61ab), it seems that the image and instruction parts will also become bidirectional?
- The paper mentions that the model receives empty action embeddings as input, but I find in [datasets.py#L57](https://github.com/moojink/openvla-oft/blob/main/prismatic/vla/datasets/datasets.py#L57) it seems to receive ground truth action id during training. And during inference [modeling_prismatic.py#L738](https://github.com/moojink/openvla-oft/blob/main/prismatic/extern/hf/modeling_prismatic.py#L738), the model indeed receives empty embedding. Do I have a misunderstanding?

**Comment by niiickZ** [comment]
Thank you for your reply. For the second question, my understanding is that computing the loss only requires the `labels` containing ground truth, while the `input_ids` can be left as empty input.

**Comment by moojink** [comment]
Hi @niiickZ,

1. Yes, all attention masks become bidirectional. In OpenVLA-OFT we only care about generating the correct actions anyway (image/language onputs have no effect on training loss).
2. The model receives empty action embeddings as input during training as well. You can see this in the forward pass logic here: https://github.com/moojink/openvla-oft/blob/213c5ff8d6e9673885d7e4dd6f2ce73b5d8a45db/prismatic/extern/hf/modeling_prismatic.py#L618
